pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -819,6 +828,52 @@ func TestIsSkipped(t *testing.T) {
 			},
 		}},
 		expected: false,
+	}, {
```

Comment:
Can we add more tests here for examples such as:

1.

```
                                           B 
                                         / 
A (guarded) (CAS=true)
                                        \ 
                                          C  
```

2. 

```
                                           B 
                                         / 
A (guarded) (CAS=true)
                                        \ 
                                          C (guarded) (CAS=true) - D 
```

3. 

```
A (guarded) (CAS=true)
                                           \
                                                C
                                           /
B                                      
```

Reply:
sounds good, will add them :)